feat: Implement wishlist feature

nayoonk928 opened the pull request:
### 변경사항
**AS-IS**

* 위시리스트에 컨텐츠 추가/삭제 기능 구현
* 사용자의 위시리스트에 있는 컨텐츠 모두 가져오는 기능 구현

**TO-BE**

### 테스트
- [x] 테스트 코드
- [x] API 테스트

nayoonk928 committed:
feat: Implement wishlist feature

- 위시리스트에 컨텐츠 추가/삭제 기능 구현
- 사용자 전체 위시리스트 내용 가져오는 기능 구현
- 테스트 코드 구현

mingming-mentor commented on src/main/java/com/personal/contentservice/service/impl/WishlistServiceImpl.java:
```diff
@@ -0,0 +1,100 @@
+    List<Wishlist> wishlists = wishlistRepository.findByUser(user);
+    List<WishlistDto.Response> wishlistDto = new ArrayList<>();
+
+    for (Wishlist wishlist : wishlists) {
+      Content content = wishlist.getContent();
+      WishlistDto.Response response = convertContentToDto(content);
+      wishlistDto.add(response);
+    }
```
stream을 이용해서 만들어주어도 될듯해요. 가독성이 좀더 높아질 듯 합니다

nayoonk928 replied:
stream 이용하여 수정하였습니다.

mingming-mentor commented on src/main/java/com/personal/contentservice/service/impl/WishlistServiceImpl.java:
```diff
@@ -0,0 +1,100 @@
+        .build();
+  }
+
+  private User getUser(Authentication authentication) {
+    PrincipalDetails principalDetails = (PrincipalDetails) authentication.getPrincipal();
+    User user = principalDetails.getUser();
+    if (user == null) {
+      throw new CustomException(USER_NOT_FOUND);
+    }
+    return user;
+  }
```
이 getUser 메소드는 여러곳에서 사용이 된다면 따로 빼두어도 좋을 듯 합니다

mingming-mentor commented on src/main/java/com/personal/contentservice/domain/User.java:
```diff
@@ -27,6 +27,10 @@ public class User extends BaseEntity {
 
+  @JsonBackReference
+  @OneToMany(mappedBy = "user")
+  private List<Wishlist> wishlists = new ArrayList<>();
```
보니까 
`List<Wishlist> wishlists = wishlistRepository.findByUser(user);`
이런식으로 user 정보로 wishList를 다시 조회해오는 형태로 되어있는듯해요. 그렇다면 User entity에서는 굳이 wishList 맵핑을 해주지 않아도 될듯합니다.

nayoonk928 replied:
감사합니다 wishlist 매핑 제거했습니다.

mingming-mentor commented on src/main/java/com/personal/contentservice/domain/User.java:
```diff
@@ -27,6 +27,10 @@ public class User extends BaseEntity {
   @OneToMany(mappedBy = "user")
   private List<Review> reviews = new ArrayList<>();
 
+  @JsonBackReference
```
요 annotation을 붙여준 이유가 있을까요?

nayoonk928 replied:
순환참조를 방지하기 위해 사용하였는데 제거하고 이 코드에서는 필요 없는 것이 확인되어 삭제하였습니다.

mingming-mentor reviewed (approved):
리뷰드렸습니다! 수고 많으셨어요

nayoonk928 committed:
refactor: refactor domain and service code

- user, content domain 에서 wishlist 매핑 코드 삭제
- getUser 메서드가 서비스 클래스들에서 공통으로 사용되므로 UserAuthenticationUtils 클래스를 생성하여 따로 작성
